File: app/src/main/java/com/lzy/nestescroll/RecyclerAdapter.java
package com.lzy.nestescroll;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;


import com.lzy.nestescroll.views.BaseViewHolder;
import com.lzy.nestescroll.views.titleview.TitleView;
import com.lzy.nestescroll.views.titleview.TitleViewViewModel;

import java.util.List;

public class RecyclerAdapter extends RecyclerView.Adapter<BaseViewHolder> {
    public List<String> data;

    public RecyclerAdapter(List<String> data) {
        this.data = data;
    }

    @NonNull
    @Override
    public BaseViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new BaseViewHolder(new TitleView(parent.getContext()));
    }

    @Override
    public void onBindViewHolder(@NonNull BaseViewHolder holder, int position) {
        holder.bind(new TitleViewViewModel(data.get(position)));
    }

    @Override
    public int getItemCount() {
        return data.size();
    }
}

File: app/src/main/java/com/lzy/nestescroll/RecyclerAdapter.kt
package com.lzy.nestescroll

import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.lzy.nestescroll.views.BaseViewHolder
import com.lzy.nestescroll.views.titleview.TitleView
import com.lzy.nestescroll.views.titleview.TitleViewViewModel

class RecyclerAdapter(var data: ArrayList<String>) :
    RecyclerView.Adapter<BaseViewHolder<TitleViewViewModel>>() {
    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): BaseViewHolder<TitleViewViewModel> {
        return BaseViewHolder(TitleView(parent.context))
    }

    override fun onBindViewHolder(
        holder: BaseViewHolder<TitleViewViewModel>,
        position: Int
    ) {
        holder.bind(TitleViewViewModel(data[position]))
    }

    override fun getItemCount(): Int {
        return data.size
    }

}